spyder/plugins/ipythonconsole/confpage.py

```diff
@@ -255,6 +238,69 @@ def setup_page(self):
         run_file_layout.addWidget(run_file_browser)
         run_file_group.setLayout(run_file_layout)
 
+        # ---- Debug ----
+        # Pdb run lines Group
+        pdb_run_lines_group = QGroupBox(_("Run code while debugging"))
+        pdb_run_lines_label = QLabel(_(
+            "You can run several lines of code on each "
+            "new prompt while debugging. Please "
+            "introduce each one separated by semicolons "
+            "and a space, for example:<br>"
+            "<i>import matplotlib.pyplot as plt</i>"))
+        pdb_run_lines_label.setWordWrap(True)
+        pdb_run_lines_edit = self.create_lineedit(
+            _("Lines:"), 'startup/pdb_run_lines', '', alignment=Qt.Horizontal)
+
+        pdb_run_lines_layout = QVBoxLayout()
+        pdb_run_lines_layout.addWidget(pdb_run_lines_label)
+        pdb_run_lines_layout.addWidget(pdb_run_lines_edit)
+        pdb_run_lines_group.setLayout(pdb_run_lines_layout)
+
+        # Debug Group
+        debug_group = QGroupBox(_("Debug"))
+        debug_layout = QVBoxLayout()
+
+        prevent_closing_box = newcb(
+            _("Prevent editor from closing files while debugging"),
+            'pdb_prevent_closing',
+            tip=_("This option prevents the user from closing a file while"
+                  " it is debugged."))
+        debug_layout.addWidget(prevent_closing_box)
+
+        continue_box = newcb(
+            _("Stop debugging on first line of files without breakpoints"), 'pdb_stop_first_line',
```

Comment:
This option is new, right? Could you explain a bit more why you added it and how it works?

Reply:
Ha yes - I wanted to write a comment about that. Because I am adding a Debugging settings tab, I added two small tweaks I am using locally but didn't have time to make into proper PR:
 - `pdb_stop_first_line`: The problem I have is that the debugger will stop on the first line if there is no breakpoints. Because it is annoying to always have to press continue, spyder will automatically do it if a breakpoint is detected. The problem I have is that it doesn't do that if no breakpoints are detected, event if the code will call other files where I have breakpoints. So I added an option to always do continue, so the user is responsible to add breakpoins before debugging.
 - `pdb_prevent_closing`: This allows me to close a file in the editor even if I am debugging it. I am quite frustrated every time spyder doen't let me close a file I asked it to close, so I like this option a lot.